How programs work (#10)
LaunchCodeEducation/data-analysis-curriculum: content/installations/install-visual-studio/_index.md

+++
title = "Visual Studio Code Installation"
date = 2024-02-12T13:42:28-06:00
draft = false
weight = 20
+++

## Setting Up Visual Studio Code

> Before we start coding on our computer, we need to make sure we have the right tools! Programmers use **integrated development environments (IDE)** to write and run their code. The development environment we will be using for this class is Visual Studio Code

Reviewer: ```suggestion
Before we start coding on our computer, we need to make sure we have the right tools! Programmers use **rich text editor** to write and run their code. The editor we will be using for this class is Visual Studio Code
```

Reviewer: Technically, Visual Studio is an IDE and Visual Studio Code is a rich text editor that is only able to run code because of installed plugins.

> Before we start coding on our computer, we need to make sure we have the right tools! Programmers use **integrated development environments (IDE)** to write and run their code. The development environment we will be using for this class is Visual Studio Code
> .

Reviewer: ```suggestion
Before we start coding on our computer, we need to make sure we have the right tools! Programmers use **integrated development environments (IDE)** to write and run their code. The development environment we will be using for this class is Visual Studio Code.
```

In addition to simply writing and running code, Visual Studio Code has tools that recognize errors in our code and has an integrated terminal so we can navigate through our filesystem to find the files that need our attention.

{{% notice blue Note "rocket" %}}
Visual Studio Code is very customizable. Once you have everything set up, you can take additional steps to personalize your workspace such as changing the theme.
{{% /notice %}}

Go to the Visual Studio Code [download page](https://code.visualstudio.com/download/) and download the appropriate version.

## Mac Users

> Please ensure that you download the correct version depending on your chip type.  If you have an _M1_ or _M2_ you are running on _Apple Silicon_.  If you have a different chip, you are most likely running on an _Intel_ chip. If you are not sure, click on the Apple icon in the top left of your screen.  Select "About this Mac".  You should see information regarding your _Chip_.  And don't hesitate to do an internet search about your chip if you are still not sure.

Reviewer: ```suggestion
Please ensure that you download the correct version depending on your chip type.  If you have an M1 or M2 you are running on Apple Silicon.  If you have a different chip, you are most likely running on an Intel chip. If you are not sure, click on the Apple icon in the top left of your screen.  Select "About this Mac".  You should see information regarding your chip.  And don't hesitate to do an internet search about your chip if you are still not sure.
```

Open your new copy of Visual Studio Code. To open one of your coding projects, go to _File > Open_ and select the project you want to work on.

> To start working with the terminal, look for the _Terminal_ tab in the top menu.  To open at new terminal,  _Terminal > New Terminal_. The new terminal window will open on the bottom of your IDE.

Reviewer: ```suggestion
To start working with the terminal, look for the _Terminal_ tab in the top menu.  To open a new terminal,  _Terminal > New Terminal_. The new terminal window will open on the bottom of your IDE.
```
